.gitlab/source_test/ebpf.yml

```diff
@@ -31,9 +31,11 @@ tests_ebpf:
     # Compile runtime security stress tests to be executed in kitchen tests
     - inv -e security-agent.build-stress-tests --output=$DD_AGENT_TESTING_DIR/site-cookbooks/dd-security-agent-check/files/stresssuite
     # Compile master version for comparison, uncomment following lines when merged
-    #- git checkout master
-    #- inv -e security-agent.build-stress-tests --output=$DD_AGENT_TESTING_DIR/site-cookbooks/dd-security-agent-check/files/stresssuite-master
-    #- git checkout -
+    - git checkout master
+    - go mod vendor
+    - inv -e system-probe.build --bundle-ebpf --incremental-build
+    - inv -e security-agent.build-stress-tests --output=$DD_AGENT_TESTING_DIR/site-cookbooks/dd-security-agent-check/files/stresssuite-master
+    - git checkout -
```

Comment:
Perhaps we should compare to the latest release in release.json ?  This makes builds less repeatable since master changes between invocations of this task.

Reply:
The problem with this approach is that, if since the last release you have one merged commit that improved the performance by 10%, if the PR introduces a 5% penalty, it won't be an error (as it still an improvement compared to the last release)

Reply:
The inverse problem exists with the approach where you use master.   If you have one PR that incurs a 5% penalty, followed by another commit that incurs a 5% penalty, neither will show up since both changes are marginally acceptable.

We should keep in mind that this task will be run as part of every release. Any issues will be on the releaser to fix. We should prevent cases where the ebpf tests can't run because  of a commit that is already merged into master.

Reply:
@leeavital just to understand, this tests are run currently manually, are they triggered during the release process ? 

About on what to compare, both approaches are acceptable, one solution could be to compare both but probably something we could introduce later once we have learnt from multiple PR comparisons

Reply:
While it may be interesting to compare the performance of a release against the previous one, doing so for every commit is hard to manage :
Let's say a PR introduced a 5% penalty and this penalty is expected because the feature has a cost.
With the approach you described, every following PR will show this 5% penalty. So every member of the team has to be aware of the current delta against the last release to be able to know if his PR introduces a regression.

What we could do is comparing against master for every PR and periodically - and at minimum for a new release - to compare against the latest release. What do you think ?